pylint/config.py

```diff
@@ -526,6 +527,28 @@ def parse(self, to_parse, config):
         pass
 
 
+def make_commenter(description):
+    """Make a function to add comments from re.sub."""
+    def make_comment(match):
+        """Make the replacement string including commented desciption."""
+        comment = '\n'.join(
+            '# {}'.format(line) for line in textwrap.wrap(description))
+        return match.expand(r'{}\n\g<0>'.format(comment))
+    return make_comment
+
+
+def make_commented_config_text(option_definitions, config_text):
+    """Make config ini text with descriptions added as comments."""
+    for name, definition in option_definitions.items():
+        config_text = re.sub(
+            r'^{}[ =]'.format(re.escape(name)),
```

Comment:
I feel like pattern matching is less safe than generating the config file with comments from the start, but can't think of a way this regex wouldn't be safe given the rules of INI files. This way the writing in `configparser` isn't reimplemented and it's easier to test just the code for comments.

Maybe if there are config options with the same name in different sections there could be a bug. Is that something to worry about?

Reply:
Not right now, but this could be possible sometimes in the future. I'm not entirely happy with how this function operates, since it is operating on the config text from the next step. I wonder if there is another way to make it work.